src/stores/EpgController.ts

```diff
@@ -102,8 +99,15 @@ export default class EpgClientService {
     } as EpgChannel;
   };
 
-  getScheduleProvider = (item: PlaylistItem) => {
-    return item?.scheduleType === EPG_TYPE.VIEW_NEXA ? this.viewNexaProvider : this.jwProvider;
+  getEpgService = (item: PlaylistItem) => {
+    const scheduleType = item?.scheduleType || EPG_TYPE.JWP;
+    const service = this.epgServices.find((service) => service.type === scheduleType);
```

Comment:
What do you think about removing the type field and just using the name registered in the DI container as the type lookup?

Reply:
Used `getNamedModule` instead of multiInject. Please have a look